lib/Map/Vector/Protomaps/ProtomapsGeojsonSource.ts

```diff
@@ -0,0 +1,319 @@
+      ring.forEach((point) => {
+        point[0] = point[0] < 0 ? point[0] + 360 : point[0];
+      })
+    );
+
+    const bufferBbox = bbox(buffer);
+
+    // Get array of all features
+    const geojsonFeatures = this.geojsonObject.features;
+
+    const pickedFeatures: Feature[] = [];
+
+    for (let index = 0; index < geojsonFeatures.length; index++) {
+      const feature = geojsonFeatures[index];
+      if (!feature.bbox) {
+        feature.bbox = bbox(feature);
+      }
+
+      // Filter by bounding box and then intersection with buffer (to minimize calls to booleanIntersects)
+      if (
+        Math.max(
+          feature.bbox[0],
+          // Wrap buffer bbox if necessary
+          feature.bbox[0] > 180 ? bufferBbox[0] + 360 : bufferBbox[0]
+        ) <=
+          Math.min(
+            feature.bbox[2],
+            // Wrap buffer bbox if necessary
+            feature.bbox[2] > 180 ? bufferBbox[2] + 360 : bufferBbox[2]
+          ) &&
+        Math.max(feature.bbox[1], bufferBbox[1]) <=
+          Math.min(feature.bbox[3], bufferBbox[3])
+      ) {
+        // If we have longitudes greater than 180 - used wrappedBuffer
+        if (feature.bbox[0] > 180 || feature.bbox[2] > 180) {
+          if (booleanIntersects(feature, wrappedBuffer))
+            pickedFeatures.push(feature);
+        } else if (booleanIntersects(feature, buffer))
+          pickedFeatures.push(feature);
+      }
+    }
+
+    // Convert pickedFeatures to ImageryLayerFeatureInfos
+    pickedFeatures.forEach((f) => {
+      const featureInfo = new ImageryLayerFeatureInfo();
+
+      featureInfo.data = f;
+      featureInfo.properties = Object.assign(
+        { [LAYER_NAME_PROP]: GEOJSON_SOURCE_LAYER_NAME },
+        f.properties ?? {}
+      );
+
+      if (
+        f.geometry.type === "Point" &&
+        typeof f.geometry.coordinates[0] === "number" &&
+        typeof f.geometry.coordinates[1] === "number"
+      ) {
+        featureInfo.position = Cartographic.fromDegrees(
+          f.geometry.coordinates[0],
+          f.geometry.coordinates[1]
+        );
+      }
+
+      featureInfo.configureDescriptionFromProperties(f.properties);
+      featureInfo.configureNameFromProperties(f.properties);
+
+      featureInfos.push(featureInfo);
+    });
+
+    return featureInfos;
+  }
+}
+
+export const geomTypeMap = (
+  type: string | null | undefined
+): GeomType | null => {
+  switch (type) {
+    case "Point":
+    case "MultiPoint":
+      return GeomType.Point;
+    case "LineString":
+    case "MultiLineString":
+      return GeomType.Line;
+    case "Polygon":
+    case "MultiPolygon":
+      return GeomType.Polygon;
+    default:
+      return null;
+  }
+};
+
+export function geojsonVtTileToProtomapsFeatures(
+  features: geojsonvt.Feature[],
+  tileSize: number
+): ProtomapsFeature[] {
+  const scale = tileSize / GEOJSON_VT_EXTENT;
+
+  return features
+    .map((f) => {
+      let transformedGeom: Point[][] = [];
+      let numVertices = 0;
```

Comment:
I don't think these two need to be initialized (looks like they are assigned in all code paths below).

Reply:
Strictly speaking no, but they make sense as default values, so i'm going to leave them there